selene_sdk/samplers/random_positions_sampler.py

```diff
@@ -151,86 +159,214 @@ def __init__(self,
-            self.interval_lengths, select_indices[:n_indices_validate])
-        self._sample_from_mode["validate"] = SampleIndices(
-            val_indices, val_weights)
+            self.chroms_info["Total"].append(chrom)
+            self.chroms_info["Starts"][counter] = tot_len
+            self.chroms_info["Ends"][counter] = tot_len + len_chrom
+            tot_len += len_chrom
+            counter += 1
+
+    # Compute the number of elements in the genome array that belong to
+    # each mode, by proportion (floored to an int)
 
+    def _assign_proportions(self):
         if self.test_holdout:
-            n_indices_test = int(n_intervals * self.test_holdout)
-            test_indices_end = n_indices_test + n_indices_validate
-            test_indices, test_weights = get_indices_and_probabilities(
-                self.interval_lengths,
-                select_indices[n_indices_validate:test_indices_end])
-            self._sample_from_mode["test"] = SampleIndices(
-                test_indices, test_weights)
-
-            tr_indices, tr_weights = get_indices_and_probabilities(
-                self.interval_lengths, select_indices[test_indices_end:])
-            self._sample_from_mode["train"] = SampleIndices(
-                tr_indices, tr_weights)
+            test_prop_N = self._genome_n_bases * self._test_holdout
+            validation_prop_N = self._genome_n_bases * self._validation_holdout
+            training_prop_N = \
+                self._genome_n_bases * (1 - self._test_holdout - self._validation_holdout)
+            return int(test_prop_N), int(validation_prop_N), int(training_prop_N)
         else:
-            tr_indices, tr_weights = get_indices_and_probabilities(
-                self.interval_lengths, select_indices[n_indices_validate:])
-            self._sample_from_mode["train"] = SampleIndices(
-                tr_indices, tr_weights)
+            validation_prop_N = self._genome_n_bases * self._validation_holdout
+            training_prop_N = self._genome_n_bases * (1 - self._validation_holdout)
+            return int(validation_prop_N), int(training_prop_N)
+
+    # Should be updated to be dynamic for the genome that is being sampled.
+    # If the genome size is not a multiple of 250 million, then we do it
+    # differently ..
+    def _psuedoshuffle(self):
+        total = self._genome_n_bases
+        shuffled_sample = np.arange(total, dtype=np.int64)
+        jump = 250000000
+        num_steps = int(total / jump)
+        start = 0
+        for i in range(num_steps):
+            np.random.shuffle(shuffled_sample[start : start + jump])
+            start = start + jump
+
+        start = int(jump / 2)
+        for i in range(num_steps - 1):
+            np.random.shuffle(shuffled_sample[start : start + jump])
+            start = start + jump
+
+        return shuffled_sample
+
+    # Samples for each mode are predetermined based on the sample size for
```

Comment:
This is a lot of documentation that is helpful but should be stuff that you move into the PR text comment vs in the code. You can make a docstring for `_partition_by_proportion` or wherever is appropriate with these kinds of comments too. It's good that you have written it though, just move it elsewhere :)

Reply:
Ya its mostly in the PR comment right now. I will cut it down to make the code easier to scroll through and maybe add it in the docstring when its complete (since the design of the shuffle may change as well..)